app/models/temporary_date_fields.rb

```diff
@@ -0,0 +1,20 @@
+module TemporaryDateFields
+  DATE_FIELDS = %i(
+    temporary_first_published_at
+    temporary_last_edited_at
+    temporary_major_published_at
+    temporary_published_at
+  ).freeze
+
+  def set_temporary_date_fields
+    DATE_FIELDS.each_with_object({}) do |attribute, hash|
+      hash[attribute] = previously_published_item.send(attribute)
+    end
+  end
+
+  def set_default_temporary_date_fields
+    DATE_FIELDS.each_with_object({}) do |attribute, hash|
+      hash[attribute] = Time.zone.now
```

Comment:
We'd probably want to avoid using Time.zone.now and instead have a date injected. Otherwise these dates which are basically the same will end up with microsecond differences which have caught people out before.